plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/wizard/dependencies/ExternalLibrariesInstallHelper.java

```diff
@@ -102,22 +112,24 @@ private void removeDependenciesToShippedCodeIfVersionMatches(Map<String, String>
 	 * *all* shipped code projects need to be shadowed, because shipped code is now published in clusters in which each
 	 * project depends on the others with a fixed version.
 	 */
-	private void addDependenciesForRemainingShippedCode(Map<String, String> dependenciesToInstall,
+	private void addDependenciesForRemainingShippedCode(Map<String, NPMVersionRequirement> dependenciesToInstall,
 			Set<String> projectNamesOfShippedCode) {
 		Set<String> projectNamesOfShippedCodeToInstall = new HashSet<>(dependenciesToInstall.keySet());
 		projectNamesOfShippedCodeToInstall.retainAll(projectNamesOfShippedCode);
 		if (!projectNamesOfShippedCodeToInstall.isEmpty()
 				&& projectNamesOfShippedCodeToInstall.size() < projectNamesOfShippedCode.size()) {
-			Set<String> versionConstraintsOfShippedCodeToInstall = projectNamesOfShippedCodeToInstall
+			Map<String, NPMVersionRequirement> versionRequirementsOfShippedCodeToInstall = projectNamesOfShippedCodeToInstall
 					.stream()
 					.map(id -> dependenciesToInstall.get(id))
-					.collect(Collectors.toSet());
-			if (versionConstraintsOfShippedCodeToInstall.size() > 1) {
-				// FIXME GH-957 warn about conflicting version constraints for shipped code!
-			}
-			String versionConstraint = versionConstraintsOfShippedCodeToInstall.stream().findFirst()
-					.orElse(null);
+					.collect(Collectors.toMap(SemverSerializer::serialize, Function.identity(), (vr1, vr2) -> vr1));
```

Comment:
Does it work with multiple version requirements that have a solution, like `>0.1.0` and `0.1.1`?

Reply:
No, this is a very simple equality check. Otherwise we'd need sophisticated logic to check if one version requirement intersects with another, one implies another, etc. That's why I added a warning in case of multiple requirements remaining.